code.js

```diff
@@ -1,3 +1,53 @@
-function binarySearch(list, element) {
-    return -1;
+function binarySearch(list = [], element) {
+
+    pivotPercentage = .5; 
+    index = -1;
+
+    // So now I'm using a percentage to place my pivot in the middle of the data and then using math to manipulate the place the index,
+    // this will place the pivot halfway in between the set of numbers I want to look at, binary search! 
+    
+    for (i = 0; i < Math.floor(list.length / 2) + 2; i++){
+        pivot = Math.floor(pivotPercentage * list.length);
+        //console.log(element, list, list[pivot], i); // uncomment if you wish to see how the algorithm is doing in runtime
+        if (list[pivot] == element) {
+            index = pivot;
+            repeatCheck = false;
+            while (repeatCheck == false){ // Testing for repeats and changing the index to the first occurance of our element
+                if  (list[pivot - 1] == element) {
+                    index = pivot - 1;
+                    pivot = pivot - 1;
+                }
+                else {
+                    repeatCheck = true;
+                }
+            }
+            return index;
+        }
+        else if (list[pivot] > element) { // Tests if the element is smaller than the pivot item
+            pivotPercentage *= .5; // Multiplying a fraction by 1/2 will make it exactly half, perfect for looking in the middle of
```

Comment:
Interesting way of doing it, but you're not splitting the list in half as binary search would.

Reply:
I see, should I make the code slice the list then? Or use a set of brackets to represent the start and end of my list without changing the index of the list?

Reply:
Anything that repeatedly halves the list.

Reply:
Ok I can do that, thanks!

Reply:
Ok now I made a proper binary search! What do you think?